client-core/src/service/wallet_service.rs

```diff
@@ -248,28 +354,76 @@ where
     }
 
     /// Get the wallet state from storage
+    // storage -> wallet
     pub fn get_wallet_state(&self, name: &str, enckey: &SecKey) -> Result<WalletState> {
         load_wallet_state(&self.storage, name, enckey)?.err_kind(ErrorKind::InvalidInput, || {
             format!("WalletState with name ({}) not found", name)
         })
     }
 
+    /// Save wallet to storage
+    pub fn save_wallet(&self, name: &str, enckey: &SecKey, wallet: &Wallet) -> Result<()> {
+        self.storage.save_secure(KEYSPACE, name, enckey, wallet)?;
+
+        let info_keyspace = get_info_keyspace(name);
+        // write viewkey
+        write_pubkey_enc(
+            &self.storage,
+            &info_keyspace,
+            "viewkey",
+            &wallet.view_key,
+            enckey,
+        )?;
+
+        for (pubkey, prikey) in &wallet.key_pairs {
+            self.add_key_pairs(&name, &enckey, &pubkey, &prikey)?
+        }
+
+        write_number(&self.storage, &info_keyspace, "publicindex", 0)?;
+        // pubkey
+        for public_key in wallet.public_keys.iter() {
+            self.add_public_key(name, enckey, public_key)?;
+        }
+
+        // stakingkey
+        write_number(&self.storage, &info_keyspace, "stakingkeyindex", 0)?;
+        for public_key in wallet.staking_keys.iter() {
+            self.add_staking_key(name, enckey, public_key)?;
+        }
+
+        // root hash
+        write_number(&self.storage, &info_keyspace, "roothashindex", 0)?;
+        for root_hash in wallet.root_hashes.iter() {
+            self.add_root_hash(name, enckey, root_hash.clone())?;
+        }
+
+        Ok(())
+    }
+
     /// Store the wallet to storage
+    // wallet -> storage
     pub fn set_wallet(&self, name: &str, enckey: &SecKey, wallet: Wallet) -> Result<()> {
-        save_wallet(&self.storage, name, enckey, &wallet)
+        self.save_wallet(name, enckey, &wallet)
     }
 
     /// Finds staking key corresponding to given redeem address
     pub fn find_staking_key(
         &self,
         name: &str,
-        enckey: &SecKey,
+        _enckey: &SecKey,
```

Comment:
perhaps TODO comment to change this API?

Reply:
ok